sdk/identity/azure-identity/azure/identity/_credentials/default.py

```diff
@@ -137,7 +140,13 @@ def __init__(self, **kwargs):
         if not exclude_powershell_credential:
             credentials.append(AzurePowerShellCredential(**kwargs))
         if not exclude_interactive_browser_credential:
-            credentials.append(InteractiveBrowserCredential(tenant_id=interactive_browser_tenant_id, **kwargs))
+            if interactive_browser_client_id:
+                credentials.append(InteractiveBrowserCredential(
+                    tenant_id=interactive_browser_tenant_id,
+                    client_id=interactive_browser_client_id,
```

Comment:
InteractiveBrowserCredential expects this argument to be set only with a valid client ID (i.e. a string). So, we should pass a value here only when interactive_browser_client_id has a string value.

Reply:
I don't see we do similar validation for other client ids. Should we keep consistent with others?

Reply:
`InteractiveBrowserCredential(client_id=None, ...)` creates a credential that doesn't work. `ManagedIdentityCredential(client_id=None)` creates a credential that could work, depending on the environment. Maybe both credentials should be consistent in how they handle `client_id=None` but they aren't today. (For what it's worth, I think InteractiveBrowserCredential is more correct here.)